stat/stat.go

```diff
@@ -24,6 +26,275 @@ const (
 	LinInterp CumulantKind = 4
 )
 
+const epsilon = 1e-8
+
+// WassersteinDistance computes the Wasserstein distance (Earth Mover's Distance)
+// between two 1D distributions p and q with optional weights.
+//
+// Parameters:
+// p and q are slices of values representing the support points of each distribution.
+// pWeights and qWeights are optional weights for each point. If nil, uniform weights are used.
```

Comment:
```suggestion
// WassersteinDistance computes the Wasserstein distance (Earth Mover's Distance)
// between two 1D distributions, p and q with optional weights. p and q are are
// the support points of each distribution. pWeights and qWeights are the weights
// for each point.
// The lengths of p and q must be equal. If a weights slice is nil then all of
// the weights are 1. If it is not nil, then the length of the weights slice
// must equal the length of the corresponding points.
```
(similar change below)